pkg/controller/kubedirectorcluster/members.go

```diff
@@ -284,7 +284,21 @@ func handleReadyMembers(
 				return
 			}
 
-			memberVersion := *m.StateDetail.LastConnectionVersion
+			var memberVersion int64
+
+			if m.StateDetail.LastConnectionVersion != nil {
+				memberVersion = *m.StateDetail.LastConnectionVersion
+
+			} else {
+				memberVersion = connectionsVersion - 1
+
+				shared.LogInfo(
+					reqLogger,
+					cr,
+					shared.EventReasonCluster,
+					fmt.Sprintf("Member Version is : %d. Was NIL", memberVersion),
```

Comment:
I'm not sure this is a useful message? If this is just a debug log message that needs to be removed, you can skip the rest of this comment.

However if it's something you think we should keep, a few tweaks are needed:
- If you just want to log this (and not generate an event), you can use shared.EventReasonNoEvent.
- Rather than building the message string yourself with fmt.Sprintf, use `shared.LogInfof` (rather than `shared.LogInfo` and pass the variables-to-be-substituted in as the final arguments. You can find examples of that elsewhere in this file.
- "Member Version" is too generic; we're specifically talking about "lastConnectionVersion" here.
- To really be a useful message we'd need the member name (from m.Pod) to be logged.

Reply:
Thanks, Joel. I've changed this to not generate an event.